sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filtering/Filter.java

```diff
@@ -0,0 +1,197 @@
+        TelemetryColumns data) {
+        // All of the filters need to match for this to return true (and operation).
+        for (FilterInfo filter : filterConjunctionGroupInfo.getFilters()) {
+            if (!checkFilter(filter, data)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static boolean checkFilter(FilterInfo filter, TelemetryColumns data) {
+        if (ANY_FIELD.equals(filter.getFieldName())) {
+            return checkAnyFieldFilter(filter, data);
+        } else if (filter.getFieldName().startsWith(CUSTOM_DIM_FIELDNAME_PREFIX)) {
+            return checkCustomDimFilter(filter, data);
+        } else {
+            Object fieldValue = data.getFieldValue(filter.getFieldName());
+
+            if (filter.getFieldName().equals(KnownRequestColumns.SUCCESS)) {
+                if (fieldValue == null) { // this should never happen, but just in case
+                    return false;
+                }
+                boolean fieldValueBoolean = (Boolean) fieldValue;
+                boolean comparand = Boolean.parseBoolean(filter.getComparand().toLowerCase());
+                if (filter.getPredicate().equals(PredicateType.EQUAL)) {
+                    return fieldValueBoolean == comparand;
+                } else if (filter.getPredicate().equals(PredicateType.NOT_EQUAL)) {
+                    return fieldValueBoolean != comparand;
+                }
+            } else if (filter.getFieldName().equals(KnownDependencyColumns.DURATION)) {
+                if (fieldValue == null) { // this should never happen, but just in case
+                    return false;
+                }
+                long comparand = getMicroSecondsFromFilterTimestampString(filter.getComparand());
+                return numericCompare((Long) fieldValue, comparand, filter.getPredicate());
+            } else if (filter.getFieldName().equals(KnownDependencyColumns.RESULT_CODE)
+                || filter.getFieldName().equals(KnownRequestColumns.RESPONSE_CODE)) {
+                if (fieldValue == null) { // this should never happen, but just in case
+                    return false;
+                }
+                int comparand = Integer.parseInt(filter.getComparand());
+                PredicateType predicate = filter.getPredicate();
+                return numericCompare((Integer) fieldValue, comparand, predicate);
+            } else {
+                // string fields
+                return stringCompare((String) fieldValue, filter.getComparand(), filter.getPredicate());
+            }
+        }
+        return false;
+    }
+
+    private static boolean checkAnyFieldFilter(FilterInfo filter, TelemetryColumns data) {
+        try {
+            Field[] fields = data.getClass().getDeclaredFields();
+            for (Field field : fields) {
+                field.setAccessible(true);
```

Comment:
With a map (https://github.com/Azure/azure-sdk-for-java/pull/43040#pullrequestreview-2458938038) you would not need to use reflection.